main/signatures/index.ts

```diff
@@ -0,0 +1,22 @@
+import { TypedMessage, TypedSignatureRequestType } from '../accounts/types'
+import signatureTypes from './types'
+import { SignTypedDataVersion, MessageTypeProperty } from '@metamask/eth-sig-util'
+
+export const identify = ({ data }: TypedMessage<SignTypedDataVersion>): TypedSignatureRequestType => {
```

Comment:
this is another great candidate for unit tests, both as an example of how it works and to verify a lot of these logical checks